azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -53,182 +75,229 @@ def initialize(self):
                 if self._initialized:
                     return
 
-                #
                 # check whether cgroup monitoring is supported on the current distro
-                #
                 self._cgroups_supported = CGroupsApi.cgroups_supported()
                 if not self._cgroups_supported:
                     logger.info("Cgroup monitoring is not supported on {0}", get_distro())
                     return
 
-                #
-                # check systemd
-                #
+                # check that systemd is detected correctly
                 self._cgroups_api = SystemdCgroupsApi()
-
                 if not self._cgroups_api.is_systemd():
-                    message = "systemd was not detected on {0}".format(get_distro())
-                    logger.warn(message)
-                    add_event(op=WALAEventOperation.CGroupsInitialize, is_success=False, message=message, log_event=False)
+                    self.__log_cgroup_warning("systemd was not detected on {0}", get_distro())
                     return
 
-                def log_cgroup_info(format_string, *args):
```

Comment:
The intialize() method was getting too long, so I refactored it into smaller methods. The new code is mostly in __ensure_azure_slices_exist